multicluster/controllers/multicluster/commonarea/acnp_resourceimport_controller.go

```diff
@@ -0,0 +1,191 @@
+				klog.ErrorS(err, msg, "acnp", klog.KObj(acnpObj))
+				return ctrl.Result{}, r.reportStatusEvent(msg, ctx, resImp)
+			}
+		} else if !apiequality.Semantic.DeepEqual(acnp.Spec, acnpObj.Spec) {
+			acnp.Spec = acnpObj.Spec
+			if err = r.localClusterClient.Update(ctx, acnp, &client.UpdateOptions{}); err != nil {
+				msg := "Failed to update imported Antrea ClusterNetworkPolicy in cluster " + r.localClusterID
+				klog.ErrorS(err, msg, "acnp", klog.KObj(acnpObj))
+				return ctrl.Result{}, r.reportStatusEvent(msg, ctx, resImp)
+			}
+		}
+	} else if tierNotFound && !acnpNotFound {
+		// The ACNP Tier does not exist, and the policy cannot be realized in this particular importing member cluster.
+		// If there is an ACNP previously created via import (which has a valid Tier by then), it should be cleaned up.
+		if err = r.localClusterClient.Delete(ctx, acnpObj, &client.DeleteOptions{}); err != nil {
+			msg := "Failed to delete imported Antrea ClusterNetworkPolicy that no longer has a valid Tier for cluster " + r.localClusterID
+			klog.ErrorS(err, msg, "acnp", klog.KObj(acnpObj))
+			return ctrl.Result{}, r.reportStatusEvent(msg, ctx, resImp)
+		}
+	} else if tierNotFound {
+		msg := fmt.Sprintf("ACNP Tier %s does not exist in importing cluster %s", tierName, r.localClusterID)
+		return ctrl.Result{}, r.reportStatusEvent(msg, ctx, resImp)
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *ResourceImportReconciler) handleResImpDeleteForClusterNetworkPolicy(ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) (ctrl.Result, error) {
+	acnpName := types.NamespacedName{
+		Namespace: "",
+		Name:      common.AntreaMCSPrefix + resImp.Spec.Name,
+	}
+	klog.InfoS("Deleting ACNP corresponding to ResourceImport",
+		"acnp", acnpName.String(), "resourceimport", klog.KObj(resImp))
+
+	acnp := &v1alpha1.ClusterNetworkPolicy{}
+	err := r.localClusterClient.Get(ctx, acnpName, acnp)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(2).InfoS("ACNP corresponding to ResourceImport has already been deleted",
+				"acnp", acnpName.String(), "resourceimport", klog.KObj(resImp))
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	if err = r.localClusterClient.Delete(ctx, acnp, &client.DeleteOptions{}); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func getMCAntreaClusterPolicy(resImp *multiclusterv1alpha1.ResourceImport) *v1alpha1.ClusterNetworkPolicy {
+	if resImp.Spec.ClusterNetworkPolicy == nil {
+		return nil
+	}
+	return &v1alpha1.ClusterNetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: common.AntreaMCSPrefix + resImp.Spec.Name,
+			Annotations: map[string]string{
+				common.AntreaMCACNPAnnotation: "true",
+			},
+		},
+		Spec: *resImp.Spec.ClusterNetworkPolicy,
+	}
+}
+
+func (r *ResourceImportReconciler) reportStatusEvent(errMsg string, ctx context.Context, resImp *multiclusterv1alpha1.ResourceImport) error {
+	if errMsg != "" {
```

Comment:
I would suggest to use below logic to return first which can help to reduce unnecessary indents. 
```
if errMsg == "" {
  return nil
}
```

Reply:
Done